File: 卡码/代码随想录/JS/二叉树/02-二叉树属性/LeetCode_0110.js
//110. 平衡二叉树
import TreeNode from "../TreeNode.js";

const getHeight=root=>{
  if(!root) return 0;
  const leftHeight=getHeight(root.left),rightHeight=getHeight(root.right);
  if(leftHeight==-1||rightHeight==-1) return -1;
  return Math.abs(leftHeight-rightHeight)>1 ? -1 : 1+Math.max(leftHeight,rightHeight);
};
const isBalanced = function(root) {
  return getHeight(root)==-1 ? false : true;
};

File: 卡码/代码随想录/Go/二叉树/02-二叉树属性/LeetCode_0110.go
//110. 平衡二叉树
package main

import (
	"math"
	. "binaryTree/types"
)

func getHeight(root *TreeNode) int{
	if root==nil {
		return 0
	}
	leftHeight:=getHeight(root.Left)
	rightHeight:=getHeight(root.Right)
	if leftHeight==-1||rightHeight==-1 {
		return -1
	}
	if math.Abs(float64(leftHeight)-float64(rightHeight))>1 {
		return -1
	}else{
		return 1+int(math.Max(float64(leftHeight),float64(rightHeight)))
	}
}

func isBalanced(root *TreeNode) bool {
	if getHeight(root)==-1 {
		return false
	}else{
		return true
	}
}
